src/Swarrot/Processor/Retry/RetryProcessor.php

```diff
@@ -61,7 +61,11 @@ public function process(Message $message, array $options)
                 $properties
             );
 
-            $key = str_replace('%attempt%', $attempts, $options['retry_key_pattern']);
+            $key = str_replace(
+                '%attempt%',
+                ($attempts > $options['retry_attempts']) ? $options['retry_attempts'] : $attempts,
```

Comment:
So for a given number of retries you want to publish with specific routing_keys (`smtg_retry_1`, `smtg_retry_2`, `smtg_retry_3`) and once the attempt limit is reached you want to always publish with a dedicated routing_key (`smtg_retry`)?